### ISSUE TITLE
Error messages "Material not defined"

### ISSUE THREAD

**Issue by michas0602** [buddhi1980/mandelbulber2#643]
##### System information (version)
- mandelbulber2 => 2.18 dev
- Operating System / Platform => Win7 64bit

##### Detailed description

 To reproduce just load a fractal with 1 material only, create a second one and delete the first.

**Comment by mclarekin** [comment]
latest dev?  Maybe Krzysztof  made some change for better handling of this
problem.

First test - i have no problem with the program working OK. but got  9
critical warnings about it

**Comment by michas0602** [comment]
Latest dev. These critical warnings already occured in the last 1 or 2 versions...however, something is wrong here

**Comment by mclarekin** [comment]
Yes those critical warnings have benn filling up my log for a while now.

Has the program crashed (mine just did), and the last info on my log were
these warnings (of course that may be due to something else)

**Comment by buddhi1980** [comment]
When you delete material which is already assigned to the object, it is normal. Material 1 by default is used for main fractal object.
![image](https://user-images.githubusercontent.com/11696990/57468842-8e972e00-7285-11e9-8354-d44040e28794.png)
 If you delete this material, you should assign new material (e.g. material 2) to main fractal object.
These messages are warning and error handling for case, if there is used material which is no longer defined. Program continues running, but material is automatically substituted to any existing material. Rendering result can be different as expected.
